Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/ProfileAWSCredentialIdentityResolver.swift

```diff
@@ -262,6 +291,24 @@ public struct ProfileAWSCredentialIdentityResolver: AWSCredentialIdentityResolve
             )
         }
     }
+
+    private func prependFeatureIDs(
+        featureIDsToPrepend: [String],
+        creds: AWSCredentialIdentity
+    ) -> AWSCredentialIdentity {
+        let originalFeatureIDs = creds.properties.get(key: AWSIdentityPropertyKeys.credentialFeatureIDs) ?? []
+        let newFeatureIDs = featureIDsToPrepend + originalFeatureIDs
+        var properties = creds.properties
+        properties.set(key: AWSIdentityPropertyKeys.credentialFeatureIDs, value: newFeatureIDs)
+        return AWSCredentialIdentity(
+            accessKey: creds.accessKey,
+            secret: creds.secret,
+            accountID: creds.accountID,
+            expiration: creds.expiration,
+            sessionToken: creds.sessionToken,
+            properties: properties
+        )
+    }
```

Comment:
Helper function that takes in feature ID list and a AWS creds object. Returns a new AWS creds object with feature IDs prepended to feature ID list that the old AWS creds object had.